gensim/models/word2vec.py

```diff
@@ -481,6 +484,7 @@ def build_vocab(
         report_values = self.prepare_vocab(update=update, keep_raw_vocab=keep_raw_vocab, trim_rule=trim_rule, **kwargs)
         report_values['memory'] = self.estimate_memory(vocab_size=report_values['num_retained_words'])
         self.prepare_weights(update=update)
+        self.add_lifecycle_event("build_vocab", update=update, trim_rule=str(trim_rule))
```

Comment:
Should we log some (or all) the parameters here?

Reply:
As a minimal internal log, I'd think *all* parameters could be unwieldy, but a major switch like `update` makes sense, and even (where not too hard) some very-high-level indicators of the magnitude of the operation, such as elapsed time and before/after model vocabulary size.

Reply:
Yes – we want to record some parameters, but not all.

If you have concrete ideas what should be recorded where, please push a suggestion / commit. Different events call for different semantics, so I don't think this can be automated.